packages/app/src/components/DBTraceWaterfallChart.tsx

```diff
@@ -391,19 +404,24 @@ export function DBTraceWaterfallChartContainer({
     const start = startOffset - minOffset;
     const end = start + tookMs;
 
-    const body = result.Body;
-    const serviceName = result.ServiceName;
-    const type = result.type;
+    const { Body: body, ServiceName: serviceName, id, type } = result;
 
-    const id = result.id;
+    // Extract HTTP-related logic
+    const eventAttributes = result.SpanAttributes || {};
+    const hasHttpAttributes =
```

Comment:
Just realizing, I think we want to condition this on spanKind client? that way it's only shown for HTTP requests from a client (since servers should be handled elsewhere in the logic iirc)